packages/elements/src/react/common/form/otp.tsx

```diff
@@ -88,173 +87,83 @@ const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps,
           }}
-          onTouchEnd={e => {
-            const isFocusing = document.activeElement === e.currentTarget;
-            if (isFocusing) {
-              _select();
-            }
-
-            rest?.onTouchEnd?.(e);
+          onSelect={event => {
+            setSelectionRange(cur => selectionRangeUpdater(cur, innerRef));
+            rest?.onSelect?.(event);
           }}
-          onTouchMove={e => {
-            const isFocusing = document.activeElement === e.currentTarget;
-            if (isFocusing) {
-              _select();
-            }
-
-            rest?.onTouchMove?.(e);
+          onMouseOver={event => {
+            setIsHovering(true);
+            props.onMouseOver?.(event);
           }}
-          onDoubleClick={e => {
-            const isFocusing = document.activeElement === e.currentTarget;
-
-            if (isFocusing) {
-              e.currentTarget.setSelectionRange(0, e.currentTarget.value.length);
-              _select();
-            }
-
-            rest?.onDoubleClick?.(e);
-          }}
-          onInput={e => {
-            _select();
-
-            rest?.onInput?.(e);
-          }}
-          onKeyDown={e => {
-            _keyDown(e);
-            _select();
-
-            rest?.onKeyDown?.(e);
-          }}
-          onKeyUp={e => {
-            _select();
-
-            rest?.onKeyUp?.(e);
-          }}
-          onFocus={e => {
-            if (innerRef.current) {
-              const start = Math.min(innerRef.current.value.length, length - 1);
-              const end = innerRef.current.value.length;
-              innerRef.current.setSelectionRange(start, end);
-              setSelectionRange([start, end]);
-            }
-
-            rest?.onFocus?.(e);
+          onMouseLeave={event => {
+            setIsHovering(false);
+            props.onMouseLeave?.(event);
           }}
           style={inputStyle}
-          data-otp-input-segmented
         />
         <div
-          className={userProvidedClassName}
+          className={className}
           aria-hidden
-          style={renderStyle}
-          data-otp-input-segmented-render
+          style={segmentWrapperStyle}
         >
-          {Array.from({ length: length }).map((_, i) => {
-            const value = String(props.value)[i] || '';
-            const isCursor = isFocused && selectionRange[0] === selectionRange[1] && selectionRange[0] === i;
-            const isSelected =
-              isFocused &&
-              selectionRange[0] !== null &&
-              selectionRange[0] <= i &&
-              selectionRange[1] !== null &&
-              selectionRange[1] > i;
-
+          {Array.from({ length }).map((_, i) => {
             return (
               <React.Fragment key={`otp-segment-${i}`}>
                 {render({
-                  value,
-                  status: isCursor ? 'cursor' : isSelected ? 'selected' : 'none',
+                  value: String(props.value)[i] || '',
+                  status:
```

Comment:
This nested ternary logic is hard too read for me so I'd prefer to keep the separation like previously

Reply:
sounds good 👍